pkg/planner/cardinality/row_count_index.go

```diff
@@ -345,16 +345,23 @@ func getIndexRowCountForStatsV2(sctx planctx.PlanContext, idx *statistics.Index,
 			histNDV := idx.NDV
 			// Exclude the TopN in Stats Version 2
 			if idx.StatsVer == statistics.Version2 {
-				c := coll.GetCol(idx.Histogram.ID)
-				// If this is single column of a multi-column index - use the column's NDV rather than index NDV
+				colIDs := coll.Idx2ColUniqueIDs[idx.Histogram.ID]
+				// Retrieve column statistics for the 1st index column
+				c := coll.GetCol(colIDs[0])
+				// If this is single column predicate - use the column's information rather than index.
+				// Index histograms are converted to string. Column uses original type - which can be more accurate for out of range
 				isSingleColRange := len(indexRange.LowVal) == len(indexRange.HighVal) && len(indexRange.LowVal) == 1
-				if isSingleColRange && !isSingleColIdx && c != nil && c.Histogram.NDV > 0 {
+				if isSingleColRange && c != nil && c.Histogram.NDV > 0 {
```

Comment:
```suggestion
				if isSingleColRange && && isSingleColIdx && c != nil && c.Histogram.NDV > 0 {
```

Do we need to keep the isSingleColIdx check here?

Reply:
/hold

Reply:
We can remove the isSingleColIdx check - because we only care that the range is single column. A single column range against a multi-column index should also use the column statistics.